src/_zkapauthorizer/tests/test_storage_server.py

```diff
@@ -524,3 +547,173 @@ def test_mutable_share_sizes(
             actual_sizes,
             Equals(expected_sizes),
         )
+
+    @given(
+        storage_index=storage_indexes(),
+        secrets=tuples(
+            write_enabler_secrets(),
+            lease_renew_secrets(),
+            lease_cancel_secrets(),
+        ),
+        test_and_write_vectors_for_shares=slot_test_and_write_vectors_for_shares(),
+    )
+    def test_mutable_spending_metrics(
```

Comment:
I would be strongly inclined to fold these tests into the existing happy path tests which live in `test_storage_protocol`, rather than having multiple tests that exercise these code paths in the same way.

Reply:
I agree the factoring here is not great.  I am somewhat inclined to leave it as-is for now, at least until the other metrics code we want to write is written and maybe until the spending service changes land, and then do a refactoring pass.